clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -10,4 +10,57 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <clang/CIR/Dialect/IR/CIRDialect.h>
+#include "clang/CIR/Dialect/IR/CIRDialect.h"
+
+#include "mlir/Support/LogicalResult.h"
+
+#include "clang/CIR/Dialect/IR/CIROpsDialect.cpp.inc"
```

Comment:
Alphabetization concerns again.

Reply:
The first of the three includes here is the header file for this source file.  In some coding standards that header should always be first.  Does Clang have that convention?

The last of the three includes is the TableGen-generated implementation file.  I feel like that should always be last.  Is there a Clang or LLVM convention about where that should go?

That leaves only one include, the middle one, that needs to be alphabetized.

Reply:
Here's our policy https://llvm.org/docs/CodingStandards.html#include-style

So tablegen files don't seem to be special cases? Current header seems like it should be first.

Reply:
The TableGen'erated file doesn't have its own includes.  So it needs to be last, to rely on the includes of the main `.cpp` file.  The TableGen file is not a regular header and shouldn't be lumped in with the guidelines for the other headers.

Reply:
> The TableGen'erated file doesn't have its own includes. So it needs to be last, to rely on the includes of the main .cpp file.

I think that should probably change (though I don't think it needs to as part of this patch) because relying on transitive includes is rarely a good thing for long-term maintenance.